File: dehaze-go/internal/model/query/login_log_query.go
package query

// LoginLogQuery 登录日志分页查询条件
type LoginLogQuery struct {
	Username   string `json:"username"`
	IP         string `json:"ip"`
	Status     *int   `json:"status"`
	DeviceType string `json:"deviceType"`
	StartTime  string `json:"startTime"`
	EndTime    string `json:"endTime"`
	// 可见用户范围（普通用户仅本人日志时由 API 层限定为 [当前用户ID]，管理员为空表示全量）
	UserIDs []int64 `json:"userIds"`
	// 页码
	PageNum int `json:"pageNum"`
	// 每页条数
	PageSize int `json:"pageSize"`
}

File: dehaze-java/src/main/java/com/pei/dehaze/model/query/LoginLogQuery.java
package com.pei.dehaze.model.query;

import com.pei.dehaze.common.base.BasePageQuery;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 登录日志分页查询对象
 */
@Schema(description = "登录日志分页查询对象")
@Data
@EqualsAndHashCode(callSuper = true)
public class LoginLogQuery extends BasePageQuery {

    @Schema(description = "按用户名筛选（精确匹配）")
    private String username;

    @Schema(description = "按IP筛选（精确匹配）")
    private String ip;

    @Schema(description = "登录状态(1:成功;0:失败)")
    private Integer status;

    @Schema(description = "设备类型(web/android/flutter/miniprogram)")
    private String deviceType;

    @Schema(description = "开始时间")
    private String startTime;

    @Schema(description = "结束时间")
    private String endTime;
}
